### PR TITLE
Threshold flow/v6

### PR DESCRIPTION
**Pull request by victorjulien**
SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1701

Replaces https://github.com/OISF/suricata/pull/10595
- rebase
- ticket link for minor todo (https://redmine.openinfosecfoundation.org/issues/6855)
- add support for detection_filter as well

https://redmine.openinfosecfoundation.org/issues/6822

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.

ERROR: QA failed on SURI_TLPR1_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 138 | 147 | 106.52% |
| | SURI_TLPR1_stats_chk | | |
| .uptime | 638 | 679 | 106.43% | 

Pipeline 19292

**Review by catenacyber — COMMENTED** [review]
Thanks for the work

CI : ✅
Code : TODO
Commits segmentation : nice (last commit adding ticket id could be squashed)
Commit messages : nice
Git ID set : looks fine for me
CLA : :-p
Doc update : ok, but `threshold.config` needs an update in its comment ;-)
Redmine ticket : ok
Rustfmt : not needed
Tests : nice SV
Dependencies added: none

File: src/detect-engine-threshold.c [hunk 1]
```diff
@@ -277,6 +342,7 @@ static int ThresholdHandlePacketSuppress(Packet *p,
             }
             break;
         case TRACK_RULE:
+        case TRACK_FLOW:
```
Comment (catenacyber) [inline src/detect-engine-threshold.c:345]: What is this line doing ? And `ThresholdHandlePacketSuppress` ?

File: src/detect-engine-register.h [hunk 2]
```diff
@@ -86,6 +83,20 @@ enum DetectKeywordId {
     DETECT_BYTETEST,
     DETECT_BYTEJUMP,
     DETECT_BYTEMATH,
+    DETECT_BYTE_EXTRACT,
+    DETECT_DATASET,
+    DETECT_DATAREP,
+    DETECT_BASE64_DECODE,
+    DETECT_BASE64_DATA,
+    DETECT_BSIZE,
```
Comment (catenacyber) [inline src/detect-engine-register.h:91]: Should `DETECT_ISDATAAT` be in here as well ?

Comment (catenacyber) [reply src/detect-engine-register.h:91]: And `DETECT_AL_URILEN` cf https://github.com/OISF/suricata/pull/10334/files#diff-20b12e868dc376bbab11d9bd32507cc49db149d15f3d2274c0697e4700351bd5

**Review by jufajardini — COMMENTED** [review]
doc changes look good to me.

**Comment by victorjulien** [comment]
> Is this the new version of #9488 ?

No, thats a different thresholding PR.

File: src/detect-engine-threshold.c (hunk 1 above)
Comment (victorjulien) [reply src/detect-engine-threshold.c:345]: it makes it explicit that TRACK_FLOW isn't supported for type suppress

File: src/detect-engine-register.h (hunk 2 above)
Comment (victorjulien) [reply src/detect-engine-register.h:91]: yes

**Comment by victorjulien** [comment]
comments addressed in #10910

**Pull request closed without merging**